common/rfb/SSecurityPlain.cxx

```diff
@@ -53,6 +55,13 @@ bool PasswordValidator::validUser(const char* username)
   for (size_t i = 0; i < users.size(); i++) {
     if (users[i] == "*")
       return true;
+#if !defined(WIN32) && !defined(__APPLE__)
+    if (users[i] == "%u") {
+      struct passwd *pw = getpwuid(geteuid());
+      if (pw && !strcmp(pw->pw_name, username))
```

Comment:
This is technically not the correct way to do things, but I guess it's an existing problem. The system should be comparing uid:s, not strings, as there might be multiple valid usernames for the same user.

Reply:
I have a better solution for this:
```
    if (users[i] == "%u") {
      struct passwd *pw = getpwnam(username);
      if (pw && pw->pw_uid == getuid())
        return true;
    }
```